CONTRIBUTING.md

```diff
@@ -175,3 +175,90 @@ For testing and development you can find self-signed certificates under
 ```bash
  openssl req -nodes -x509 -newkey rsa:4096 -keyout key.pem -out cert.pem -sha256 -days 365 -subj '/CN=localhost'
 ```
+
+## Logging Principles
+Logging plays a crucial role for administrators, users and developers of our tool, which is why it
+is important to have certain standards in regards on when and how to use which type of logger.
+
+The following sections will outline
+- what information should be logged.
+- when information should be logged.
+- how information should be logged.
+
+### What information should be logged.
+#### Unknown-Error Log Message
+These types of messages are things such as an unexpected exception was thrown, which should
+provider relative information and context on the exception.
+The following questions should be answered with these log messages.
+- What error happened
+- what data was used to cause this error
+- when did the error happen
```

Comment:
The list does not have to be exhaustive, but should start clear and concise. Hence, I would start with:

```
1. Why was the exception thrown by DataCater?
2. What parameter led to the Exception? In e.g. `URIInvalidException`, because of invalid String for an URI, then the String should be printed out. [No user information is allowed to be printed out]
```
That would be my suggestion. Feel free to suggest another list. Let us avoid `- ...` to avoid ambiguity.